lib/scripting/objects/pinball-backend/rom-fetcher.ts

```diff
@@ -0,0 +1,122 @@
+import { GamelistDB } from 'wpc-emu';
+import { logger } from '../../../util/logger';
+
+/**
+ * Functions to fetch a WPC ROM file from VPDB.io
+ */
+
+export function downloadGameEntry(pinmameGameName: string): Promise<LoadedGameEntry> {
+	const gameEntry: undefined | GamelistDB.ClientGameEntry = GamelistDB.getByPinmameName(pinmameGameName);
+	if (!gameEntry) {
+		return Promise.reject(new Error('GAME_ENTRY_NOT_FOUND_' + pinmameGameName));
+	}
+	const url: string = buildVpdbGameEntryUrl(gameEntry.pinmame.vpdbId || gameEntry.pinmame.id);
+	return downloadFileAsJson(url)
+		.then((jsonData: VpdbGameEntry[]) => {
+			if (!Array.isArray(jsonData)) {
+				return Promise.reject(new Error('VPDB_INVALID_ANSWER'));
+			}
+			const result: VpdbGameEntry | void = jsonData.find((vpdbEntry: VpdbGameEntry) => vpdbEntry.id === pinmameGameName);
+			if (!result) {
+				return Promise.reject(new Error('VPDB_GAME_ENTRY_NOT_FOUND'));
+			}
+			logger().debug(pinmameGameName, 'VPDB RESULT:', jsonData);
+
+			const romSet: VpdbGameEntry | void = findRomSet(jsonData, pinmameGameName);
+			if (!romSet) {
+				return Promise.reject(new Error('VPDB_ROMSET_ENTRY_NOT_FOUND'));
+			}
+			logger().debug(pinmameGameName, 'VPDB romSet:', romSet);
+
+			const romName: string | undefined = findMainRomFilename(romSet);
```

Comment:
Remove type

Reply:
hmm whats better then when i remove the type here? (sorry, dump question - but I don't understand it - on the other hand it's my first ts code, there might be a lot I don't understand)

Reply:
It doesn't add any value, TypeScript knows the type because `findMainRomFilename()` defines it, and it pushes the important part (the right-hand side of the assignment) further to the right, making it less readable / skimmable.